.travis.yml

```diff
@@ -28,7 +28,7 @@ env:
         - NUMPY_VERSION=stable
         - MAIN_CMD='python setup.py'
         # Build against github source until asdf-2.0 has been released
-        - ASDF_GIT='git+git://github.com/spacetelescope/asdf.git#egg=asdf'
+        - ASDF_GIT='git+git://github.com/drdavella/asdf@no-astropy-extension#egg=asdf'
```

Comment:
Wouldn't it make sense to wait until these changes are merged into asdf master before merging this PR, now that you know it works?

Reply:
This actually can be merged without any changes to ASDF, and I can revert the change to the travis config here. However, it won't actually be testing the entry point mechanism since ASDF still imports Astropy extensions as a special case. But if I eliminate the special case in ASDF before this PR is merged to Astropy, then every other Astropy PR will fail since the extension is not being loaded automatically anymore.

Reply:
@astrofrog I just rebased with the change you requested. This should be good for merge once CI passes.